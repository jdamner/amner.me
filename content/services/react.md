---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
