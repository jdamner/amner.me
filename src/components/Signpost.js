import React, { Component, Fragment } from "react";
import ReactModal from "react-modal";

import Corner from '../assets/corner1.svg';

export default class Signpost extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        }
    }

    modalStyle = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            transform: 'translate(-50%, -50%)',
            border: '4px solid black',
            zIndex: '200',
          },
        overlay: {
            backgroundColor: 'rgba(0, 0, 0, 0.75)',
        }
    }

    handleModalClick = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    handleModalClose = () => {
        this.setState({
            isOpen: false
        });
    }

    render = () => {
        ReactModal.setAppElement('#___gatsby')
        return (
            <Fragment>
                <button className='signpost' onClick={this.handleModalClick} >
                    <div className='signpost-top'></div>
                    <div className='signpost-content'>
                        <span>Contact</span>
                    </div>
                    <div className='signpost-bottom'></div>
                </button>
                <ReactModal
                    isOpen={this.state.isOpen}
                    onRequestClose={this.handleModalClose}
                    contentLabel="Contact"
                    style={this.modalStyle}
                >
                    <section className='contact' id='contact'>
                        <Corner />
                        <div className='contact-title'>
                            <h2>Say Hello</h2>
                            <p>Looking for someone just like me?</p>
                        </div>
                        <div className='contact-list'>
                            <a className='btn' href='tel:+447515352631'>
                                +44 07515 352631
                            </a>
                            <a className='btn' href='mailto:jdamner@me.com'>
                                jdamner@me.com
                            </a>
                        </div>
                    </section>
                </ReactModal>
            </Fragment>
        )
    }
}