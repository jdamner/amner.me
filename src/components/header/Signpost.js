import React, { Component, Fragment } from "react";
import { Link } from "gatsby";
import ReactModal from "react-modal";

import Corner from '../../assets/corner1.svg';

import Style from "./Style";

export default class Signpost extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        }
    }

    handleModalClick = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    handleModalClose = () => {
        this.setState({
            isOpen: false
        });
    }

    render = () => {
        return (
            <Fragment>
                <button className='signpost' onClick={this.handleModalClick} >
                    <div className='signpost-top'></div>
                    <div className='signpost-content'>
                        <span>Contact</span>
                    </div>
                    <div className='signpost-bottom'></div>
                </button>
                <ReactModal
                    appElement={document.getElementById('___gatsby')}
                    isOpen={this.state.isOpen}
                    onRequestClose={this.handleModalClose}
                    contentLabel="Contact"
                    style={Style}
                >
                    <section className='contact' id='contact'>
                        <Corner />
                        <div className='contact-title'>
                            <h2>Say Hello</h2>
                            <p>Looking for someone just like me?</p>
                        </div>
                        <div className='contact-list'>
                            <a href='tel:+447515352631'>
                                +44 07515 352631
                            </a>
                            <a href='mailto:jdamner@me.com' className=''>
                                jdamner@me.com
                            </a>
                        </div>
                    </section>
                </ReactModal>
            </Fragment>
        )
    }
}