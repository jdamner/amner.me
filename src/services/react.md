---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.