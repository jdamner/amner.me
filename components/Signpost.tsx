"use client";

import React from "react";
/* API */
import { useState } from "react";
import {
  motion,
  AnimatePresence,
  useReducedMotion,
  type HTMLMotionProps,
} from "framer-motion";

/* Components */
import Button from "./Atoms/Links/Button";

export default function Signpost({
  title,
  children,
}: {
  children?: React.ReactNode;
  title: string;
}) {
  const [isOpen, setIsOpen] = useState(false);
  const reducedMotion = useReducedMotion();

  const handleModalClick = () => {
    setIsOpen(!isOpen);
  };

  const motionProps: HTMLMotionProps<"div"> = reducedMotion
    ? {}
    : {
        initial: { y: 100, opacity: 0 },
        animate: { y: 0, opacity: 1 },
        exit: { y: 100, opacity: 0 },
        transition: { duration: 0.15, type: "spring" },
      };

  return (
    <>
      <Button onClick={handleModalClick} data-testid="contact-button">
        {title}
      </Button>
      <AnimatePresence>
        {isOpen && (
          <div
            className="fixed z-10"
            aria-labelledby="modal-title"
            role="dialog"
            aria-modal="true"
          >
            <div className="fixed inset-0 bg-gray-900 bg-opacity-75 transition-opacity">
              <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
                <motion.div
                  {...motionProps}
                  className="fixed overflow-hidden sm:my-8 sm:w-full sm:max-w-lg bg-white dark:bg-slate-800 text-slate-800 dark:text-slate-100"
                >
                  {/* Close Icon */}
                  <div className="absolute top-0 right-0 pt-4 pr-4">
                    <button
                      type="button"
                      className="active:outline-none focus:outline-none"
                      onClick={handleModalClick}
                    >
                      <span className="sr-only">Close</span>
                      <svg
                        className="w-6 h-6"
                        xmlns="http://www.w3.org/2000/svg"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                        aria-hidden="true"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          d="M6 18L18 6M6 6l12 12"
                        />
                      </svg>
                    </button>
                  </div>
                  {/* Content */}
                  {children}
                </motion.div>
              </div>
            </div>
          </div>
        )}
      </AnimatePresence>
    </>
  );
}
