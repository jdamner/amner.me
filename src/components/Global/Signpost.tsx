import React from "react";
import ReactModal from "react-modal";
import Button from "../Links/Button";
import { motion } from "framer-motion";
import { event } from "../../api/Insights";
import { useState } from "react";

export default function Signpost({ children }): JSX.Element {
    const [isOpen, setIsOpen] = useState(false);

    const modalStyle = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            transform: 'translate(-50%, -50%)',
            border: '4px solid black',
            zIndex: '200',
        },
        overlay: {
            backgroundColor: 'rgba(0, 0, 0, 0.75)',
        }
    }

    const handleModalClick = () => {
        setIsOpen(!isOpen);
        event('signpost-click', { open: !isOpen });
    }

    const handleModalClose = () => {
        setIsOpen(false);
    }

    const ContentWrapper = (props, children) => {
        return (
            <motion.div
                {...props}
                initial={{ top: -50 }}
                animate={{ top: 300 }}
                exit={{ top: -50 }}
                transition={{ duration: 0.1, type: 'spring', stiffness: 100 }}
            >
                {children}
            </motion.div>
        )
    }

    return (
        <>
            <Button onClick={handleModalClick}>Contact</Button>
            <ReactModal
                isOpen={isOpen}
                onRequestClose={handleModalClose}
                contentLabel="Contact"
                style={modalStyle}
                contentElement={ContentWrapper}
            >
                {children}
            </ReactModal>
        </>
    )
}