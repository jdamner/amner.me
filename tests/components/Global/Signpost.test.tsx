import { render } from "@testing-library/react"
import Signpost from "../../../src/components/Global/Signpost"

describe("Signpost", () => {
  it("renders correctly", () => {
    const { container } = render(<Signpost><p>Empty</p></Signpost>)
    expect(container).toMatchSnapshot()
  })
})
