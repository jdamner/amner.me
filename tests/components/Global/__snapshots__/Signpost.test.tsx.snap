// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signpost renders correctly 1`] = `
<div>
  <button
    class="border-solid border-2 border-slate-800 uppercase font-black font-sans bg-white px-4 py-2 text-slate-800 hover:bg-slate-100 hover:text-black"
  >
    Contact
  </button>
</div>
`;
